hazardous/metrics/_concordance_index.py

```diff
@@ -0,0 +1,523 @@
+import warnings
+from collections import Counter, defaultdict
+
+import numpy as np
+from scipy.interpolate import interp1d
+
+from .._ipcw import KaplanMeierIPCW
+from ..utils import check_y_survival
+
+
+def concordance_index_incidence(
+    y_test,
+    y_pred,
+    y_train=None,
+    ipcw_estimator="km",
+    time_grid=None,
+    taus=None,
+    event_of_interest=1,
+    tied_tol=1e-8,
+):
+    r"""Time-dependent concordance index for prognostic models with competing risks \
+    using inverse probability of censoring weighting.
+
+    This C-index is event-specific: we set the event of interest as a parameter,
+    and all other event labels are considered to be competing events –except 0, which
+    is the censoring event.
+
+    For example, if ``y_test`` has the following ``"event"`` columns: ``[1, 0, 2, 4]``,
+    and the event of interest is set to ``2``, then we consider the following events:
+    ``[2, 0, 1, 2]``, where 0, 1, 2 respectively denotes the censoring event, the event
+    of interest and competing events.
+
+    We define this C-index as:
+
+    .. math::
+
+        \mathrm{C}(t) = \frac{\sum_{i=1}^n \sum_{j=1}^n (\tilde{A}_{ij}
+        \hat{W}_{ij, 1}^{-1} + \tilde{B}_{ij} \hat{W}_{ij, 2}^{-1}) Q^{ij}(t)
+        \tilde{N}^1_i(t)}
+        {\sum_{i=1}^n \sum_{j=1}^n (\tilde{A}_{ij}
+        \hat{W}_{ij, 1}^{-1} + \tilde{B}_{ij} \hat{W}_{ij, 2}^{-1}) \tilde{N}^1_i(t)}
```

Comment:
I would drop the \tilde altogether to make it easier to read. Also the observed time-to-event and event are without tilde (eg T_i, T_j, D_i, D_j, ...) further down in the docstring.